azure-keyvault/azure/keyvault/custom/key_vault_authentication.py

```diff
@@ -227,8 +227,8 @@ def auth_callback(server, resource, scope, scheme):
         self.auth = KeyVaultAuthBase(authorization_callback)
         self._callback = authorization_callback
         
-    def signed_session(self):
-        session = requests.Session()
+    def signed_session(self, session=None):
+        session = session or super(KeyVaultAuthentication, self).signed_session()
```

Comment:
If you user `super` here, you need to restore the __init__ call in the __init__ (which I see is commented). Most simple here will be:
```python
session = session or requests.Session()
```

Reply:
Or change the base class to simply `Authentication`, since its init isn't doing anything (not like `OAuthTokenAuthentication`)